tests/run-make/c-link-to-rust-staticlib/rmake.rs

```diff
@@ -0,0 +1,20 @@
+// This test checks that C linking with Rust does not encounter any errors, with a static library.
+// See https://github.com/rust-lang/rust/issues/10434
+
+//@ ignore-cross-compile
+
+use run_make_support::{cc, extra_c_flags, extra_cxx_flags, run, rustc, static_lib};
+use std::fs;
+
+fn main() {
+    rustc().input("foo.rs").run();
+    cc().input("bar.c")
+        .input(static_lib("foo"))
+        .out_exe("bar")
+        .args(&extra_c_flags())
+        .args(&extra_cxx_flags())
```

Comment:
I find it surprising that the original test has `EXTRACXXFLAGS`, given that we're compiling a C library not a C++ library. Could you remove this `extra_cxx_flags()`? We can try merging and see if it's actually required.